src/Base/Stream.h

```diff
@@ -256,6 +256,101 @@ class BaseExport TextInputStream: public Stream
     std::ostringstream _ss;
 };
 
+/**
+ * The TextOutputStream class provides writing of ASCII data to an ostream, with custom handling
+ * for std::string to make it easier to write a single multi-line (or multi-word) string. This is
+ * designed for easy compatibility with the LinkStage3 implementation of the OutputStream class,
+ * used to store StringHashers for the toponaming mitigation technique.
+ */
+class BaseExport TextOutputStream: public Stream
+{
+public:
+    /** Constructor
+     * @param rin: upstream input
+     */
+    explicit TextOutputStream(std::ostream& rout)
+        : _out(rout)
+    {}
+
+    TextOutputStream(const TextOutputStream&) = delete;
+
+    TextOutputStream(const TextOutputStream&&) noexcept = delete;
+
+    void operator=(const TextOutputStream&) = delete;
+
+    void operator=(const TextOutputStream&&) = delete;
+
+    ~TextOutputStream() override = default;
+
+    template<typename T>
+    TextOutputStream& operator<<(T object)
+    {
+        _out << object << '\n';
+        return *this;
+    }
+
+    template<>
+    TextOutputStream& operator<<(const char* object)
+    {
+        std::string_view str(object);
+
+        // Count the lines so that we can deal with potential EOL conversions by external software
+        uint32_t lineCount = 0;
+        for (const auto character : str) {
+            if (character == '\n') {
+                ++lineCount;
+            }
+        }
+        // Stores the line count followed by a colon as the delimiter. We don't use
+        // whitespace because the input stream may also start with whitespace.
+        _out << lineCount << ':';

```

Comment:
Isn't it a bit weird behaviour to store the number of lines of a passed string? I guess it's needed for the element map stuff but this makes the class quite useless for any other purpose

Reply:
Good idea, I will add those tests.